Upgrade diesel from 0.16 -> 1.x

erikjohnston opened the pull request:
See [diesel changelog](https://github.com/diesel-rs/diesel/blob/master/CHANGELOG.md) for details about the API changes.

Admittedly this is mainly motivated because I was getting confused at the old `on_conflict` syntax compared with newer versions of diesel.

(I also had to add `#![feature(never_type)]` to make rfcbot compile at all).

erikjohnston commented:
Looks like the test failures is due to failing to `cargo install cargo-update`, which I don't think is my fault?

anp commented:
Hi, thanks so much!

Sorry about broken CI, I believe the issue here is being caused by a couple of issues that should be resolved in https://github.com/anp/rfcbot-rs/pull/232. Once that passes and lands you should be able to rebase onto master and get CI happy again.

anp commented on Cargo.toml:
```diff
@@ -12,8 +12,8 @@ hyper = "0.10"
-r2d2 = "0.7.1"
-r2d2-diesel = "0.16"
+r2d2 = "0.8.2"
+r2d2-diesel = "1.0.0"
```
Reading the changelog you linked (super helpful), looks like you may want to double check this: https://github.com/diesel-rs/diesel/blob/master/CHANGELOG.md#added-4

Centril commented:
#232 is now merged :)

erikjohnston commented:
I've rebased and replaced r2d2, and yay! The tests pass! Thank you both.

I forgot to mention that currently the diesel macros are causing a load of warnings during compilation, due to https://github.com/diesel-rs/diesel/issues/1785. The fix has already landed, but hasn't been released yet. We can suppress these warnings for now by adding `#![allow(proc_macro_derive_resolution_fallback)]`, but I'm not sure if we'd want to keep the warnings on to remind us that things will break if we forget to upgrade diesel again. What do people think?

anp commented:
@Centril do you have a strong opinion about how to handle the warnings from diesel?

erikjohnston commented:
> @Centril do you have a strong opinion about how to handle the warnings from diesel?

I'm guessing by the tumbleweed it might be a no? :innocent:

Centril commented:
I have no opinions at all :)

@anp please merge this if you are satisfied with it =P
